src/commands/settings/index.ts

```diff
@@ -1,28 +1,30 @@
-import {
-  ChatInputCommandInteraction,
-  GuildMember,
-  PermissionFlagsBits,
-  SlashCommandBuilder
-} from 'discord.js';
-import { Command } from '../../utilities/command';
-import { karma } from './karma';
-import { scanMembers } from './scan-members';
-import { setup } from './setup';
+import type { ChatInputCommandInteraction, GuildMember } from 'discord.js';
+import { PermissionFlagsBits } from 'discord.js';
+
+import { Command } from '../../types/command.js';
+import type {
+  CommandBuilder,
+  CommandBuilderOutput
+} from '../../types/command-builder.js';
+import { karma } from './karma.js';
+import { scanMembers } from './scan-members.js';
+import { setup } from './setup.js';
 
 // TODO: Not implemented/not validated subcommands:
 // setup: Once we have more than one setting for a guild, we can use setup as a wizard to guide admins through them
 // scan-members: If/when Rusty gets a website, this command will iterate through all existing users to make sure they have a document in firestore
 
-export default class SettingsCommand extends Command {
-  async build() {
-    return new SlashCommandBuilder()
-      .setName('settings')
-      .setDescription('Configuration options')
-      .addSubcommand((setup) =>
-        setup.setName('setup').setDescription('First time setup wizard')
+export class SettingsCommand extends Command {
+  public readonly name = 'settings';
+  public readonly description = 'Configuration options';
+
+  override build(commandBuilder: CommandBuilder): CommandBuilderOutput {
+    return commandBuilder
+      .addSubcommand((subcommand) =>
+        subcommand.setName('setup').setDescription('First time setup wizard')
       )
-      .addSubcommand((karma) =>
-        karma
+      .addSubcommand((subcommand) =>
+        subcommand
           .setName('karma')
```

Comment:
Is there a way that we can have subcommands dictate their own name and description too?

Reply:
Probably, but def not in this PR. Its a whole can of worms building structs based on that. Will revisit later if that's alright